processing/src/main/java/org/apache/druid/query/filter/DruidPredicateFactory.java

```diff
@@ -27,6 +27,13 @@
 {
   Predicate<String> makeStringPredicate();
 
+  default Predicate<Object> makeObjectPredicate()
+  {
+    // default to try to use string predicate;
+    final Predicate<String> stringPredicate = makeStringPredicate();
+    return o -> stringPredicate.apply((String) o);
```

Comment:
uh, it is sort of built on hope right now... the hope that it will probably be a string because the only context it will currently happen in is from making vector matcher object processors on string expression columns, but if we opened it up to additional non-primitive number types like array expressions and complex column types then this would need to be reconsidered, or stronger checks in place that the filter supports filtering on the underlying object type. I might need to think about this a bit deeper.

Reply:
I think it's worth designing this properly now, so we don't have to go back and fix it later.

Maybe the right approach is to _not_ add an extra method to DruidPredicateFactory, and instead have the callers call makeStringPredicate, because the callers are the ones that should have the knowledge that it's definitely a String column being dealt with.

Would that work?

Reply:
So, I agree that String objects should just use the string predicate, but I also liked the idea of having an object predicate so that we can potentially match array types and possibly complex types.

I've modified the default implementation of object predicate to instead mimic the behavior of the `NilVectorValueMatcher` predicate, which supplied null to the string predicate.

`VectorValueMatcherColumnProcessorFactory.makeObjectProcessor ` now checks the type, and uses either `StringObjectVectorValueMatcher` which always uses the string predicate, or `ObjectVectorValueMatcher` which calls the object predicate, and so has the behavior of the `NilVectorValueMatcher`.